civicaseextras.php

```diff
@@ -134,34 +136,81 @@ function civicaseextras_civicrm_entityTypes(&$entityTypes) {
   _civicaseextras_civix_civicrm_entityTypes($entityTypes);
 }
 
-function civicaseextras_civicrm_alterAngular(\Civi\Angular\Manager $angular) {
-  _civicaseextras_civicrm_alterAngular($angular);
+function civicaseextras_civicrm_alterAngular(AngularManager $angular) {
+  _civicaseextras_alterAngular_addVisualAlert($angular);
+  _civicaseextras_alterAngular_appendOutcomePanel($angular);
 }
 
-// --- Functions below this ship commented out. Uncomment as required. ---
+/**
+ * Replaces the date column type on case lists so it displays an alert when the
+ * given date is overdue.
+ *
+ * @param $angular AngularManager
+ */
+function _civicaseextras_alterAngular_addVisualAlert (AngularManager &$angular) {
+  $changeSet = AngularChangeSet::create('display_warning_for_overdue_cases')
+    ->alterHtml('~/civicase/case/list/directives/case-list-table.directive.html',
+      function (phpQueryObject $doc) {
+        $doc->find('[ng-switch-when="date"]')
+          ->html('
+            <span ng-if="!item.overdueDates[header.name]">
+              {{ CRM.utils.formatDate(item[header.name]) }}
+            </span>
+            <strong ng-if="item.overdueDates[header.name]"
+              class="text-danger">
+              {{ CRM.utils.formatDate(item[header.name]) }}
+              <i class="material-icons civicase__icon">error</i>
+            </strong>
+          ');
+      });
+  $angular->add($changeSet);
+}
 
 /**
- * Implements hook_civicrm_preProcess().
+ * Appends the case outcomes to the case details summary.
  *
- * @link http://wiki.civicrm.org/confluence/display/CRMDOC/hook_civicrm_preProcess
+ * @param $angular AngularManager
+ */
+function _civicaseextras_alterAngular_appendOutcomePanel (AngularManager &$angular) {
+  $changeSet = AngularChangeSet::create('inject_case_outcomes')
+    ->alterHtml('~/civicase/case/details/summary-tab/case-summary-custom-data.html',
+      function (phpQueryObject $doc) {
+        $doc->find('civicase-masonry-grid')
+          ->prepend('<civicase-extras-case-outcome case="item"></civicase-extras-case-outcome>');
+      });
+  $angular->add($changeSet);
+}
+
+/**
+ * Implements hook_civicrm_alterContent().
+ * Adds extra settings fields to the Civicase Admin Settings form.
  *
-function civicaseextras_civicrm_preProcess($formName, &$form) {
+ * @link https://docs.civicrm.org/dev/en/latest/hooks/hook_civicrm_alterContent/
+ */
+function civicaseextras_civicrm_alterContent (&$content, $context, $templateName, $form) {
+  $isViewingTheCaseAdminForm = get_class($form) === CRM_Admin_Form_Setting_Case::class;
```

Comment:
Lets keep the `alterContent` hook clean. What I mean is, lets move the different jobs to different functions.
```
function civicaseextras_civicrm_alterContent () {
  addCivicaseAdminSettingsForm();
  // we can add more in future.
}

function  addCivicaseAdminSettingsForm() {}